### DISCUSSION TITLE
Spatie\Translatable attachment modal

### DISCUSSION THREAD

**Discussion by gotzsys** [filamentphp/filament#9458] [Help]
Hello everyone,

I'm using Spatie\Translatable in my model, but in the modal to attach the record, the value of the JSON appears and not the translated value. What is the correct way to work around this? Would it be through ->recordSelect()?

Tks

![image](https://github.com/filamentphp/filament/assets/13874850/18ff2dc2-8b75-4a45-8880-4effd2facd7a)

![image](https://github.com/filamentphp/filament/assets/13874850/7e99db47-c848-4c1a-ab48-45783f0089a7)

**Comment by leandrocfe** [comment]
```php
AttachAction::make()
    ->recordTitle(fn ($record) => $record->name)
```
?

**Reply by NGavanozov** [reply]
Add this function to the Model class:

```
public function setAttribute($key, $value)
    {
        if (in_array($key, $this->translatable) && !is_array($value)) {
            return $this->setTranslation($key, static::getLocale(), $value);
        }
        return parent::setAttribute($key, $value);
    }
```

Аnd that to the "ModelResouce" class:

`public ?string $activeLocale = null;`

if there is a problem with changing the language.
